lib/pbench/test/unit/agent/task/test_move_results.py

```diff
@@ -6,7 +6,7 @@
 import responses
 from click.testing import CliRunner
 
-from pbench.cli.agent.commands import results
+from pbench.cli.agent.commands.results import move
```

Comment:
Seems fine to me.  That is how it was done previously, so this change was trying to be minimal and not rototill the approach.

Reply:
IIRC, this was "latent code" which means that it doesn't hold much power of precedent.

We normally use `from` to import functions or classes (or global variables); here we are using it to import a _module_...why not use a conventional bare `import` for that?

Reply:
> IIRC, this was "latent code" which means that it doesn't hold much power of precedent.

Aren't the other modules that use the same approach considered the precedent?

Is this change really required to make this PR acceptable?  The mechanism in place does not appear to change how the code operates or its correctness.  It seems just to be mechanism that is important to be consistent about going forward for code maintainability, would you agree?  As such can we agree to not introduce a new way to do things for all other commands in this PR?  Wouldn't that kind of change be done as a separate PR addressing all the commands?

Reply:
I'm mostly concerned with determining "best practice" and encouraging its use.  I'm not opposed to having separate PRs which make existing code conform to our expectations.  But, that doesn't mean that we should accept new or changed code which does not conform (the fact that we intend to fix them later doesn't mean that it's OK to merge in departures now).

That said, my understanding of the syntax in use here (and, I grant, the change you made does follow the previous usage -- it's not a change which makes things "worse" -- but I don't think it's the form that we want) is that it requests the import of a module from a package.  Part of my objection is that I don't think that target of the `from` actually is a package in the typical Python sense.  Nevertheless, what we're interested in here is importing an entire, specific module (and not a subset of something else), and therefore we should be using a bare `import` and not the `from` form.

What is your concern about switching away from `from`, here?

Reply:
I don't want to slowly skew the code "best practices".  I'd rather have all the code behaving consistently, and then, when we decide that we want to change the usage pattern to an agreed upon best practice, move the code that way together.

Reply:
If it were a question of preserving existing practice vs. a gradual slew, I'd be willing to be persuaded, but there's not a lot of consistency currently.

I found only about a dozen cases where we are importing _modules_ using the `from` form; I found three cases 
where we use the bare `import` form; and, in the vast majority of cases (>80%), we use the `from` form targeting definitions _within_ a module.

So, I guess "our practice" is to use the `from` form to import the specific definitions that we need, which doesn't sound bad, but, in the case of test programs, we should probably be importing the whole module (because we should be testing more or less everything in it).  If we're going to import the whole module, I'd recommend importing it using a bare `import`, because I think that that syntax better expresses that intention.

Reply:
I've re-read this thread a number of times, and I think I understand now what you are trying to say.  I leaped to a different conclusion from the discussion which was not correct.

Let's what folks think about the `from` import of the leaf module.

Reply:
My initial response to that is, why did we name these functions `main()`, when doing so is bound to cause problems like this?  😉 

I think the Pythonic response is to use the `as` clause:  `from foo.bar.baz.move import main as move_baz`.